apm-agent-core/src/test/java/co/elastic/apm/agent/util/PackageScannerTest.java

```diff
@@ -45,4 +52,44 @@ void testScanJar() throws Exception {
     void getClassNamesOfNonExistentPackage() throws Exception {
         assertThat(PackageScanner.getClassNames("foo.bar", ElasticApmAgent.getAgentClassLoader())).isEmpty();
     }
+
+    @Test
+    void threadInterruptSafe() throws InterruptedException, IOException {
+
+        CountDownLatch end = new CountDownLatch(1);
+
+        URL testJar = Test.class.getProtectionDomain().getCodeSource().getLocation();
+        try (final URLClassLoader classLoader = new URLClassLoader(new URL[]{testJar})) {
+
+            Thread t = new Thread() {
+                @Override
+                public void run() {
+
+                    // intentionally interrupt the thread, which leaves the thread in "interrupted" state
+                    interrupt();
+
+                    try {
+                        assertThat(PackageScanner.getClassNames(Test.class.getPackageName(), classLoader)).isNotEmpty();
+                    } catch (Exception e) {
+                        throw new RuntimeException(e);
+                    }
+
+                    assertThat(Thread.interrupted())
```

Comment:
avoids clearing the interrupt
```suggestion
                    assertThat(isInterrupted())
```